### ISSUE TITLE
mrcnn_nchw.uff.fp16.engine is not detecting all instances unlike mrcnn_nchw.uff.fp32.engine which is detecting all.

### ISSUE THREAD

**Issue by omi9y** [NVIDIA/TensorRT#1628]
Hi,

I have trained the MaskRCNN model which is capable of detecting 1000's instances in an image.
When converted to the TRT engine file we are facing two issues which are as follows:
1. It is not able to detect all the images like [Matterport repository](https://github.com/matterport/Mask_RCNN) is detecting.
2. Also, while converting if I'm using fp32 mode then I can detect not all but a considerable number of instances in the images but as soon as I switch to fp16 mode then only the top few instances of the image is getting detected.

I'm using jetson xavier nx and sampleUFFMaskRcnn code to detect the instances.

**Comment by oxana-nvidia** [comment]
Hi @omi9y ,
Could you please share more details about TensorRT and cuda version that you are using? Also if possible please provide examples of the images that are not working correctly. 

Also we encourage to use the following template when filing an issue:

TensorRT Version:
NVIDIA GPU:
NVIDIA Driver Version:
CUDA Version:
CUDNN Version:
Operating System:
Python Version (if applicable):
Tensorflow Version (if applicable):
PyTorch Version (if applicable):
Baremetal or Container (if so, version):

About the issue, my first guess would be fp16 range is not enough to cover values of the weights (and possibly activations). That leads to accuracy issues in the result. TensorRT 8.4 is checking it and you can see a warning like this one when running the sample with fp16 option: 

`
./sample_uff_mask_rcnn --fp16
...
[05/25/2022-14:40:10] [W] [TRT] Weights [name=mrcnn_mask/convolution + mrcnn_mask/BiasAdd + PWN(mrcnn_mask/Sigmoid).bias] had the following issues when converted to FP16:
[05/25/2022-14:40:10] [W] [TRT]  - Subnormal FP16 values detected.
[05/25/2022-14:40:10] [W] [TRT] If this is not the desired behavior, please modify the weights or retrain with regularization to reduce the magnitude of the weights.
`

**Comment by oxana-nvidia** [comment]
Closing for now due to >14 days with no response. Please feel free to reopen if the issue still exists. Thanks
